add doc for spack install with external modules

gknapp1 opened the pull request:
I've added documentation for installing the `additivefoam` spack module when you are using modules external to spack. This is useful when you need to install OpenFOAM on the system in addition to the AdditiveFOAM solver and you have system modules available to you for common libraries like C++ and MPI.

streeve commented on docs/installation.md:
````diff
@@ -19,6 +19,57 @@ spack install additivefoam
+
+```shell
+module load gcc/8.1.0
+spack compiler add $(dirname $(which gcc))
````
Does this actually depend on the module setup? Couldn't you do this anywhere?

streeve commented on docs/installation.md:
````diff
@@ -19,6 +19,57 @@ spack install additivefoam
+spack compiler add $(dirname $(which gcc))
+```
+
+Second, you must get the location prefix of the OpenMPI installation by copying the output of the below command.
````
Any MPI?

streeve commented on docs/installation.md:
```diff
@@ -19,6 +19,57 @@ spack install additivefoam
 
+### Spack installation with external modules
+
+If you are installing OpenFOAM through spack, some configuration is needed to enable spack to use external modules, which is common on HPC systems. For OpenFOAM, this is largely needed for the C++ compiler and MPI dependencies, which are both generally available as preconfigured modules that can be loaded for functionality that has been configured for a specific HPC system. While module versions  ...
```
I would revise to say this example will use a standard HPC module setup, but the general idea of using an existing compiler/MPI can work anywhere

streeve reviewed (commented):
I could be wrong, but I think this is more general: "spack install with existing compiler/dependencies"

gknapp1 commented:
I addressed the above comments. It's a good point that this could be more general. One of the other use cases that I didn't think of originally is that if there is a system module for OpenFOAM-10, then you can create a spack package specification to wrap it and still use the spack installation of additivefoam.

I'd guess it will be less straightforward to use a standalone install of OpenFOAM-10 with spack, becuase of the environmental setup that would need to be run when loading the spack module. But I don't know enough about spack to write an example of that right now.

streeve commented:
@colemanjs good to merge?
